### ISSUE TITLE
wierd excavator bug?

### ISSUE THREAD

**Issue by yaezpk** [BluSunrize/ImmersiveEngineering#2221]
Either I should go buy a lotto ticket or it's a bug. (Just a standard coal vein)

http://imgur.com/a/TErhD

**Comment by BluSunrize** [comment]
Coal Veins are designed to contain emeralds as well. Did you remove coal ore from that chest, or is that the distribution that you got?

**Comment by yaezpk** [comment]
I didn't remove anything, thats exactly what it gave me lol

**Comment by malte0811** [comment]
Confirmed. I'm looking into this now.

**Comment by malte0811** [comment]
Fixed in my local instance. I feel like I am currently fixing the weirdest bugs ever. Also it seems like I was extremely lucky to be able to reproduce this:
I build a setup calling `getTileEntity` on an unloaded chunk a few days ago to try to debug #2116 (It popped up on FC and I saw the setup on one of Soaryn's streams). It seems like MC tries to generate a village every time that chunk is "semi-loaded". Village generation calls this method in the world with parameters only dependent on the chunk as far as I can see:
```java
/**
 * puts the World Random seed to a specific state dependant on the inputs
 */
public Random setRandomSeed(int p_72843_1_, int p_72843_2_, int p_72843_3_)
{
   long i = (long)p_72843_1_ * 341873128712L + (long)p_72843_2_ * 132897987541L + this.getWorldInfo().getSeed() + (long)p_72843_3_;
   this.rand.setSeed(i);
   return this.rand;
}
```
I have no idea what they are doing that for, probably to make worlds with the same seed look the same. Since the seed is the same every time the excavator uses the world random to determine the next ore the next ore is also the same most of the time. I've fixed the call I was exploiting to try to reproduce the FC scenario, I'm going to go over some of the other calls to fix more ghostloading. I also replaced the world random with a custom random for this call.
I feel like ghostloading could be exploited to make other random things become non-random as well...
